internal/controller/system/module.go

```diff
@@ -34,21 +34,27 @@ func NewFXModule(configuration ModuleConfiguration) fx.Option {
 			meterProvider metric.MeterProvider,
 			tracerProvider trace.TracerProvider,
 		) *DefaultController {
-			var parser ledgercontroller.NumscriptParser = ledgercontroller.NewDefaultNumscriptParser()
-			if configuration.NumscriptInterpreter {
-				parser = ledgercontroller.NewInterpreterNumscriptParser(configuration.NumscriptInterpreterFlags)
-			}
+			var machineParser ledgercontroller.NumscriptParser = ledgercontroller.NewDefaultNumscriptParser()
+			var interpreterParser ledgercontroller.NumscriptParser = ledgercontroller.NewInterpreterNumscriptParser(configuration.NumscriptInterpreterFlags)
 
 			if configuration.NSCacheConfiguration.MaxCount != 0 {
-				parser = ledgercontroller.NewCachedNumscriptParser(parser, ledgercontroller.CacheConfiguration{
+				machineParser = ledgercontroller.NewCachedNumscriptParser(machineParser, ledgercontroller.CacheConfiguration{
+					MaxCount: configuration.NSCacheConfiguration.MaxCount,
+				})
+				interpreterParser = ledgercontroller.NewCachedNumscriptParser(interpreterParser, ledgercontroller.CacheConfiguration{
```

Comment:
If you think the new interpreter does not need this caching mechanism, feel free to remove it.
It is useful on the existing machine, but I don't know for the interpreter.